doc/radosgw/multisite.rst

```diff
@@ -0,0 +1,710 @@
+  instances. There will be one Zone that should be designated as the master zone
+  in a zonegroup, which will handle all bucket and user creation.
+
+- **Zonegroup**: A zonegroup consists of multiple zones, this approximately
+  corresponds to what used to be called as a region in pre Jewel releases for
+  federated deployments. There should be a master zonegroup that will handle
+  changes to the system configuration.
+
+- **Zonegroup map**: A zonegroup map is a configuration structure that holds the
+  map of the entire system, ie. which zonegroup is the master, relationships
+  between different zonegroups and certain configurables like storage policies.
+
+- **Realm**: A realm is a container for zonegroups, this allows for separation
+  of zonegroups themselves between clusters. It is possible to create multiple
+  realms, making it easier to run completely different configurations in the same
+  cluster.
+
+- **Period**: A period holds the configuration structure for the current state
+  of the realm. Every period contains a unique id and an epoch. A period's epoch
+  is incremented on every commit operation. Every realm has an associated
+  current period, holding the current state of configuration of the zonegroups
+  and storage policies. Any configuration change for a non master zone will
+  increment the period's epoch. Changing the master zone to a different zone
+  will trigger the following changes:
+  - A new period is generated with a new period id and epoch of 1
+  - Realm's current period is updated to point to the newly generated period id
+  - Realm's epoch is incremented
+
+Pool Configuration
+==================
+
+About this guide
+================
+
+In this guide we create a single zone group, with three separate zones, which
+actively sync data between them. For the purpose of the guide, we create 2 zones
+in the same cluster, and the third zone in a different cluster. There is no sync
+agent involved for mirroring data changes between the RadosGWs and this allows
+for a much simpler configuration scheme and active-active configurations. Please
+note that metadata operations such as creating a user would still need to go
+through the master zone, however data operations such as creation of buckets
+objects etc. can be handled by any of the zones.
+
+System Keys
+===========
+
+While configuring zones, RadosGW, expects creation of zone a system user, with
+an S3 like access and secret keys. This allows another radosgw instance to pull
+the configuration remotely, with the access and secret keys. For making
+scripting and use of configuration management tools easier, it would be easier
+to generate the access-key and secret-key before hand and use these. For the
+purpose of the guide we assume an access key and secret key are set in the
+environment. For eg., something like::
+
+  $ SYSTEM_ACCESS_KEY=1555b35654ad1656d805
+  $ SYSTEM_SECRET_KEY=h7GhxuBLTrlhVUyxSPUKUV8r/2EI4ngqJxD7iBdBYLhwluN30JaT3Q==
```

Comment:
Can you add an example how to create these keys? Would be bad if people just use the sample from the docs because it is too complicated to find out how to build your own.